R/pmcmc_tools.R

```diff
@@ -157,6 +184,76 @@ pmcmc_combine <- function(..., samples = list(...)) {
                 predict, chain, iteration)
 }
 
+pmcmc_combine_nested <- function(..., samples = list(...)) {
```

Comment:
I think that this can be merged with the pmcmc_combine function with a few tweaks; checkinf for the last dimension, switching on a few combine functions. You may have already tried this?

Reply:
I could probably combine them....there would still be maybe 10 lines of code in conditionals though, and not sure if it would help debugging in future.